pkg/crossplane/instance.go

```diff
@@ -151,3 +155,58 @@ func (cp *Crossplane) GetInstance(ctx context.Context, instanceID string) (*comp
 	}
 	return nil, ErrInstanceNotFound
 }
+
+// UpdateInstanceSLA updates the SLA of an instance specified by the supplied planID.
+// Only SLA changes are allowed, any other change is not permitted and yields an error.
+func (cp *Crossplane) UpdateInstanceSLA(ctx context.Context, instanceID, serviceID, planID string) error {
+	instance, err := cp.GetInstance(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+
+	instanceLabels := instance.GetLabels()
+	if serviceID != instanceLabels[ServiceIDLabel] {
+		return ErrServiceUpdateNotPermitted
+	}
+
+	newPlan, err := cp.GetPlan(ctx, planID)
+	if err != nil {
+		return err
+	}
+
+	instanceSLA := instanceLabels[SLALabel]
+	newPlanSLA := newPlan.Labels[SLALabel]
+
+	slaChangePermitted := func() bool {
+		if instanceSLA == SLAPremium && newPlanSLA == SLAStandard {
+			return true
+		}
+		if instanceSLA == SLAStandard && newPlanSLA == SLAPremium {
+			return true
+		}
+		return false
+	}
+
+	if !slaChangePermitted() {
+		return ErrSLAChangeNotPermitted
```

Comment:
According to the spec, this should return a `422` error code. So we probably should return an APIError here?

https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#response-5

Reply:
you're absolutely right.. and I had it once (not returning an APIError but handling the errors on the caller side) but I removed it for some reason. Will re-add that.